lxd/images.go

```diff
@@ -1989,6 +1990,108 @@ func imageExport(d *Daemon, r *http.Request) response.Response {
 	return response.FileResponse(r, files, nil, false)
 }
 
+func imageExportPost(d *Daemon, r *http.Request) response.Response {
+	project := projectParam(r)
+	fingerprint := mux.Vars(r)["fingerprint"]
+
+	// Check if the image exists
+	_, _, err := d.cluster.ImageGet(project, fingerprint, false, true)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	req := api.ImageExportPost{}
+
+	err = json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		response.SmartError(err)
+	}
+
+	// Connect to the target and push the image
+	args := &lxd.ConnectionArgs{
+		TLSServerCert: req.Certificate,
+		UserAgent:     version.UserAgent,
+		Proxy:         d.proxy,
+		CachePath:     d.os.CacheDir,
+		CacheExpiry:   time.Hour,
+	}
+
+	// Setup LXD client
+	remote, err := lxd.ConnectLXD(req.Target, args)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	var imageCreateOp lxd.Operation
+
+	run := func(op *operations.Operation) error {
+		createArgs := &lxd.ImageCreateArgs{}
+		imageMetaPath := shared.VarPath("images", fingerprint)
+		imageRootfsPath := shared.VarPath("images", fingerprint+".rootfs")
+
+		metaFile, err := os.Open(imageMetaPath)
+		if err != nil {
+			return err
+		}
+		defer metaFile.Close()
+
+		createArgs.MetaFile = metaFile
+		createArgs.MetaName = filepath.Base(imageMetaPath)
+
+		if shared.PathExists(imageRootfsPath) {
+			rootfsFile, err := os.Open(imageRootfsPath)
+			if err != nil {
+				return err
+			}
+			defer rootfsFile.Close()
+
+			createArgs.RootfsFile = rootfsFile
+			createArgs.RootfsName = filepath.Base(imageRootfsPath)
+		}
+
+		image := api.ImagesPost{
+			Filename: createArgs.MetaName,
+			Source: &api.ImagesPostSource{
+				Fingerprint: fingerprint,
+				Secret:      req.Secret,
+			},
```

Comment:
Shouldn't we have a `Mode`, mirroring what's done for instances, having it default to `pull` but require it be set to `push` in this case?